embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/comms/delivery/EmbraceDeliveryCacheManager.kt

```diff
@@ -181,26 +211,16 @@ internal class EmbraceDeliveryCacheManager(
      * If loadObject returns null, it tries to load the old version of the file which was storing
      * a list of [PendingApiCall] instead of [PendingApiCalls].
      */
-    override fun loadPendingApiCalls(): PendingApiCalls {
-        logger.logDeveloper(TAG, "Loading pending api calls")
-        val callable = Callable<PendingApiCalls?> {
-            val loadApiCallsResult = runCatching {
-                cacheService.loadObject(PENDING_API_CALLS_FILE_NAME, PendingApiCalls::class.java)
-            }
-            loadApiCallsResult.getOrNull() ?: loadPendingApiCallsOldVersion()
-        }
-        val cached = backgroundWorker.submit(callable = callable).get()
-        return if (cached != null) {
-            cached
-        } else {
-            logger.logDeveloper(TAG, "No pending api calls cache found")
-            PendingApiCalls()
-        }
-    }
+    override fun loadPendingApiCalls(): PendingApiCalls =
+        runCatching { cacheService.loadObject(PENDING_API_CALLS_FILE_NAME, PendingApiCalls::class.java) }.getOrNull()
+            ?: loadPendingApiCallsOldVersion()
+            ?: PendingApiCalls()
 
     override fun replaceSession(sessionId: String, mutator: (SessionMessage) -> SessionMessage) {
         val filename = cachedSessions[sessionId]?.filename ?: return
-        cacheService.replaceSession(filename, mutator)
+        backgroundWorker.submit {
```

Comment:
This can be removed as the invocation of `replaceSession` happens on a background thread

Reply:
Ah... I saw the immediate caller not wrapped in a task, but I didn't see that method's caller wrapping this in a task. Will remove.